src/act.core.etl/Gatherer.cs

```diff
@@ -430,20 +430,32 @@ private static string ToQueryString(IDictionary<string, string> nvc)
         {
             return "?" + string.Join("&", nvc.Select(p => $"{p.Key}={p.Value}"));
         }
-        
-        
+
+
         public async Task<int> PurgeInactiveNodes()
         {
             var date = DateTime.Today.AddDays(-7);
             var count = 0;
-            _logger.LogDebug($"Getting Nodes Deactivated prior to {date.ToShortDateString()}");
+            _logger.LogInformation($"Getting Nodes Deactivated prior to {date.ToShortDateString()}");
+
+            var nodes = await _ctx.Nodes.Inactive().Where(p => p.DeactivatedDate < date)
+                .Select(p => new { p.Fqdn, p.DeactivatedDate, p.BuildSpecification.Id })
+                .ToArrayAsync();
+
+            if (nodes.Length > 0)
```

Comment:
Use Any instead of Length > 0
```suggestion
            if (nodes.Any())
```

Reply:
Any() is used for collections of IEnumerable. Length is faster because it doesn't involve creating an iterator for the array.
Since its an Array here,i prefered using Length Property as it is light-weight than any method call.